packages/nestjs-user/src/services/user-password.service.ts

```diff
@@ -36,14 +44,18 @@ export class UserPasswordService implements UserPasswordServiceInterface {
     protected readonly userLookupService: UserLookupServiceInterface,
     @Inject(PasswordCreationService)
     protected readonly passwordCreationService: PasswordCreationServiceInterface,
+    @Inject(USER_MODULE_SETTINGS_TOKEN)
+    protected readonly userSettings: UserSettingsInterface,
```

Comment:
adding all of the user password strength and roles logic directly in this service is too much, and too tightly coupled. should be done similar to password history service.